content/post/writing-a-book-remotely.md

```diff
@@ -0,0 +1,576 @@
+
+![Providing Feedback via GitHub Pull Requests](/img/stratospheric-ebook/github-pull-request-suggestion-example.png#center "Providing Feedback via GitHub Pull Requests")
+
+This way of working felt way more convenient than using a PDF reader, making comments and annotations to the file and then sending it over to the owner of the changes.
+
+After the review is completed by both other co-authors (six eye principle), we integrated the change into the `preview` branch to collect multiple chapters for the next release. Once every two month or so we then created the next work in progress release by merging the changes to the `main` branch.
+
+What's left to do is to click the release button inside Leanpub to build a new digital version of our eBook. Theoretically, we can even automate this and publish a new release on each commit for the `main` branch.
+
+## Getting the First Readers aka. Early Feedback 📖
+
+(having something to show, basic landing page, getting a cover, trouble with the naming, first version on Leanpub)
+
+Given the fact that we figured out how to sell the eBook and how to write it, we now had to start writing the eBook.
+
+We naturally found how to split the work and who write what based on our own interests. While Björn and I were more focussed on implementing features for the sample application and connecting various AWS services, Tom was more into the CI/CD and infrastructure setup.
+
+While we had a rough idea and an early draft for our planned table of contents, we adjusted our plans along the way.
+
+Working on a new chapter/feature usually involved some background fact checking, consulting the AWS documentation and other blogs, developing the code changes and then writing the chapter.
+
+For some concrete examples, these are chapters we worked on:
+- Getting Started with the AWS CDK
+- OAuth2 Login with AWS Cognito
+- Storing Entities to PostgreSQL using RDS
+- Structured Logging with Amazon CloudWatch
+- etc.
+
+While we had implemented similar features in the past, we still spend a good amount of time reading the AWS documentation and related blog posts to ensure our previous assumptions are correct.
+
+We also made sure every co-author works independently on their own area (like OAuth2 login or RDS setup) to avoid any misunderstanding.
+
+Everybody was working on a dedicated chapter so there wasn't any conflict or idle time due to waiting on some else input. Writing the new chapter consisted both of doing some research, adjusting the sample application and writing the manuscript.
+
+As were incrementally releasing new version of th eBook, we tried to work on the modules and chapters in sequence. It should be more or less and append-only of chapters and no inline adding of chapters so that the readers that get a copy early, can continue reading the book were they left if there's a new version available.
```

Comment:
```suggestion
As were incrementally releasing new version of the eBook, we tried to work on the modules and chapters in sequence. It should be more or less an append-only of chapters and no inline adding of chapters so that the readers that get a copy early, and can continue reading the book where they left off, if there's a new version available.
```